### ISSUE TITLE
Builder hut losing Items

### ISSUE THREAD

**Issue by cinjo10** [ldtteam/minecolonies#4717]
### Minecolonies version 0.11.827-ALPHA
Part of Just not enough mods 2 Mod pack

Version: 

### Expected behavior
 Builder hut page two take items from my inventory and add to its own so that the builder can use.

### Actual behavior
Builder Hut takes items from my inventory but does not add to its own

### Steps to reproduce the problem 
1. start building any building (had happen with level 1 building hut and level 1 mine)
2. used page two of builder hut to give all items needed for current build before builder requested (he was clearing area). 
3. Went to work on something else while builder worked.
4. notice the builder was requesting something, went to see what he need. It was one of the items that I had already given when I checked in hut chest it wasn't there and nether was the rest of the stuff I had just given the hut.

The way it seems to happen is to use page two of hut before the build requests the items. I used creative mode and used page two and it did keep the items a second time after builder started requesting items.

**Comment by TheKrowi** [comment]
I've noticed something similar too on version minecolonies-1.12.2-0.11.811-BETA-universal. You fulfill all needed items and suddenly the builder needs more. I've also noticed the builder getting rid of items in some way that they're deleted not to be found in the builder's hut or a rack.

Good example on my last citizen's hut build is that he needs 1 bed and actually needed to provide a bed at 3 different times in the build process (level 4 -> 5).
